File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
    	int row = 0, col = matrix[0].length-1;
			boolean flag = false; 
				while(row<=matrix.length-1 && col >=0)
				{
					if(matrix[row][col]==target)
					{
						return true;
					}
					else if(matrix[row][col]<target)
					{
						row++;
					}
					else if(matrix[row][col]>target)
					{
						col--;
					}
				}
			return flag;
    }
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        rows, cols = len(matrix), len(matrix[0])
        for i in range(rows):
            if matrix[i][0] > target:
                continue
            if matrix[i][-1] < target:
                continue
            low = 0
            high = cols - 1
            while low <= high:
                mid = (low + high) // 2
                if matrix[i][mid] == target:
                    return True
                elif matrix[i][mid] < target:
                    low = mid + 1
                else:
                    high = mid - 1
        return False
        